irods_specific_implementation/irods_rest_logical_path_api_implementation.h

```diff
@@ -115,10 +117,429 @@ namespace irods::rest
+                    msg << "- The object you tried to replicate may have a checksum that wasn't verified.\n";
+                    msg << "- You may have tried to replicate a data object onto an already identical replica.\n";
+                    msg << "Please check for these issues and contact your administrator if the issue isn't resolved.\n";
+                    return make_error_response(SYS_NOT_ALLOWED, msg.str());
+                }
+                else if ( status < 0 ) {
+                    return make_error_response(status, "Client experienced an error processing your trim request. Please check the logs or notify your administrator.");
+                }
+              }
+
+                return std::make_tuple(
+                    Pistache::Http::Code::Ok,
+                    ""
+                );
+            }
+            catch (const irods::exception& e) {
+                error("Caught exception - [error_code={}] {}", e.code(), e.what());
+                return make_error_response(e.code(), e.what());
             }
+            catch (const std::exception& e) {
+                error("Caught exception - {}", e.what());
+                return make_error_response(SYS_INVALID_INPUT_PARAM, e.what());
+            }
+	    catch (...){
+		error("Caught unknown exception.");
+		return make_error_response(SYS_UNKNOWN_ERROR, "Caught unknown exception");
+	    }
         }
+
+    private:
+        void setup_input_for_trim(
+                const Pistache::Http::Request& _request,
+                dataObjInp_t* trim_input,
+                rcComm_t& conn,
+                bool& is_collection
+            )
+        {
+              const auto _logical_path = _request.query().get("logical-path").getOrElse("");
+              is_collection = fscli::is_collection(
+                fscli::status(conn, decode_url(_logical_path))
+              );
+
+              // optional params
+              // ---------------
+              // Boolean
+              const auto _recursive   = _request.query().get("recursive").getOrElse("0");
+              // Boolean
+              const auto _dryrun      = _request.query().get("dryrun").getOrElse("");
+              // Integer
+              const auto _age         = _request.query().get("age").getOrElse("");
+              // Integer
+              const auto _num         = _request.query().get("num").getOrElse("");
+              // Integer
+              const auto _repl_num     = _request.query().get("repl-num").getOrElse("");
+              // String
+              const auto _src_resc     = _request.query().get("src-resc").getOrElse("");
+              // Boolean
+              const auto _admin       = _request.query().get("admin").getOrElse("");
+
+              if (to_bool(_recursive)) {
+                addKeyVal(
+                    &trim_input -> condInput,
+                    RECURSIVE_OPR__KW,
+                    ""
+                );
```

Comment:
There are better tools for setting key-value pairs.

See the following:
- https://github.com/irods/irods_library_examples#irods-key-value-proxy
- https://github.com/irods/irods/blob/main/lib/core/include/irods/key_value_proxy.hpp

Reply:
That looks and feels way better!